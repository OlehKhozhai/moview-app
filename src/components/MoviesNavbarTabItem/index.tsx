import React from 'react';
import cn from 'classnames';

import styles from './styles.module.scss';

type MoviesNavbarTabItemProps = {
  title: string;
  isActive: boolean;
  className?: string;
  onTabClick: (tab: string) => void;
};

const MoviesNavbarTabItem: React.FC<MoviesNavbarTabItemProps> = ({
  title,
  isActive,
  className,
  onTabClick,
}) => {
  const handleTabClick = () => {
    onTabClick(title);
  };

  return (
    <li
      className={cn(styles.root, className, {
        [styles.active]: isActive,
      })}
      onClick={handleTabClick}
    >
      {title}
    </li>
  );
};

export default React.memo(MoviesNavbarTabItem);
