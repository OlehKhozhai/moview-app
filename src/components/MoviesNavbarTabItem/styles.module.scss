@import 'assets/styles/mixins/colors';

.root {
  margin-right: 30px;
  text-transform: uppercase;
  cursor: pointer;
  padding: 20px 0;
  border-bottom: 2px solid transparent;
}

.active {
  border-bottom: 2px solid $primary-border;
}
