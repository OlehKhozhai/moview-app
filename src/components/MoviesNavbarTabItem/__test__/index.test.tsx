import React from 'react';
import { create } from 'react-test-renderer';

import MoviesNavbarTabItem from '../index';

describe('MoviesNavbarTabItem component', () => {
  test('should render MoviesNavbarTabItem component by default', () => {
    const component = create(
      <MoviesNavbarTabItem title="test" isActive={false} onTabClick={() => undefined} />
    );

    expect(component.toJSON()).toMatchSnapshot();
  });

  test('should render active MoviesNavbarTabItem component', () => {
    const component = create(
      <MoviesNavbarTabItem title="test" isActive={true} onTabClick={() => undefined} />
    );

    expect(component.toJSON()).toMatchSnapshot();
  });

  test('should trigger onTabClick method', () => {
    const onTabClickStub = jest.fn();
    const component = create(
      <MoviesNavbarTabItem title="test" isActive={true} onTabClick={onTabClickStub} />
    );

    component.root.findByType('li').props.onClick();

    expect(onTabClickStub).toHaveBeenCalledWith('test');
  });
});
