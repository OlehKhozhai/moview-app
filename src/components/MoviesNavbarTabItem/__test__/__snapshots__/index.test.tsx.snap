// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MoviesNavbarTabItem component should render MoviesNavbarTabItem component by default 1`] = `
<li
  className="root"
  onClick={[Function]}
>
  test
</li>
`;

exports[`MoviesNavbarTabItem component should render active MoviesNavbarTabItem component 1`] = `
<li
  className="root active"
  onClick={[Function]}
>
  test
</li>
`;
